src/main/java/core/basesyntax/service/impl/FruitTransactionParserImpl.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.TransactionParser;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitTransactionParserImpl implements TransactionParser {
+    private static final String SPLIT_DELIMITER = ",";
+    private static final int INDEX_FOR_OPERATION = 0;
+    private static final int INDEX_FOR_NAME = 1;
+    private static final int INDEX_FOR_QUANTITY = 2;
+
+    @Override
+    public List<FruitTransaction> converterToTransactions(List<String> readData) {
+        List<String[]> collect = readData.stream()
+                .skip(1)
+                .map(s -> s.split(SPLIT_DELIMITER))
+                .toList();
+
+        List<FruitTransaction> transactionList = new ArrayList<>();
+        FruitTransaction fruitTransaction;
+        for (String[] strings : collect) {
+            String fruitName = strings[INDEX_FOR_NAME];
+            int quantity = Integer.parseInt(strings[INDEX_FOR_QUANTITY]);
+            String operationType = strings[INDEX_FOR_OPERATION]
+                    .substring(strings[INDEX_FOR_OPERATION].length() - 1);
+            fruitTransaction =
+                    new FruitTransaction(fruitName, quantity, getOperationType(operationType));
+            transactionList.add(fruitTransaction);
+            if (quantity < 0) {
+                throw new RuntimeException("Quantity of fruit can not be less zero: "
+                        + "\""
+                        + quantity
+                        + "\"");
+            }
+        }
+        return transactionList;
+    }
+
+    private FruitTransaction.Operation getOperationType(String operationType) {
+        return switch (operationType) {
+            case "b" -> FruitTransaction.Operation.BALANCE;
+            case "r" -> FruitTransaction.Operation.RETURN;
+            case "p" -> FruitTransaction.Operation.PURCHASE;
+            case "s" -> FruitTransaction.Operation.SUPPLY;
```

Comment:
Instead of this, you need to create a method inside of enum, for-each iterate through enum values and compare, throw IllegalArgument if nothing is found. Here you are hardcoding enum codes, which is bad, you make this method tightly coupled to enum - if the enum was to be modified, such as by adding a new operation or changing the code for an operation, the getOperationType method would also need to be updated to reflect these changes. This creates a risk of errors if changes are made in one place and not the other.

By moving the logic of converting a string to an enum instance into the enum itself, through a method that iterates over all enum values and finds the corresponding one, you decouple the getOperationType method from the specific enum codes. The enum becomes responsible for knowing its own string representations, which encapsulates this knowledge within the enum itself.

If the enum provides a method like fromCode(String code), any part of the code that needs to convert a string to the enum can use this method, and if changes are necessary, they're only done within the enum. This follows the principle of single responsibility and makes the system more maintainable.